pkg/controller/workload/job/job_webhook.go

```diff
@@ -110,13 +113,20 @@ func (w *JobWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.
 func validateUpdate(oldJob, newJob *batchv1.Job) error {
 	suspendPath := field.NewPath("job", "spec", "suspend")
 
-	if queueName(oldJob) == "" && queueName(newJob) != "" && !*newJob.Spec.Suspend {
-		return field.Forbidden(suspendPath, "suspend should be true when adding the queue name")
+	if queueName(oldJob) != queueName(newJob) {
+		if queueName(oldJob) == "" && !*newJob.Spec.Suspend {
+			return field.Forbidden(suspendPath, "suspend should be true when adding the queue name")
+		}
+		if !*newJob.Spec.Suspend {
+			return field.Forbidden(suspendPath, "should not update queue name when job is unsuspend")
+		}
+		if value, exists := newJob.Annotations[constants.QueueAnnotation]; exists {
```

Comment:
instead of rewriting this code, call `validateCreate(newJob)`

Reply:
Calling `validateCreate` means validating the `parent-annotation` each time, but we already test that it is immutable, so it is enough to test it at creation.  WDYT?

Reply:
Calling both functions is the general pattern we follow in k8s validation. Right now it's relatively easy to see what needs to be validated in update without calling validateCreate, but it might be harder in the future.